components/browser/state/src/test/java/mozilla/components/browser/state/reducer/BrowserStateReducerKtTest.kt

```diff
@@ -0,0 +1,226 @@
+
+        browserState = browserState.updateTabOrCustomTabState("tab1") {
+            it.createCopy(content = it.content.copy(url = updatedUrl))
+        }
+
+        assertEquals(2, browserState.tabs.size)
+        assertEquals(updatedUrl, browserState.tabs[0].content.url)
+        assertEquals(initialUrl, browserState.tabs[1].content.url)
+    }
+
+    @Test
+    fun `GIVEN a private tab exists WHEN updateTabOrCustomTabState is called with the id of that tab THEN update the tab`() {
+        val tab1 = TabSessionState(id = "tab1", content = ContentState(url = initialUrl, private = true))
+        val tab2 = TabSessionState(id = "tab2", content = ContentState(url = initialUrl, private = true))
+        var browserState = BrowserState(tabs = listOf(tab1, tab2))
+
+        browserState = browserState.updateTabOrCustomTabState("tab1") {
+            it.createCopy(content = it.content.copy(url = updatedUrl))
+        }
+
+        assertEquals(2, browserState.tabs.size)
+        assertEquals(updatedUrl, browserState.tabs[0].content.url)
+        assertEquals(initialUrl, browserState.tabs[1].content.url)
+    }
+
+    @Test
+    fun `GIVEN a normal custom tab exists WHEN updateTabOrCustomTabState is called with the id of that tab THEN update the tab`() {
+        val tab1 = CustomTabSessionState(id = "tab1", content = ContentState(url = initialUrl), config = mock())
+        val tab2 = CustomTabSessionState(id = "tab2", content = ContentState(url = initialUrl), config = mock())
+        var browserState = BrowserState(customTabs = listOf(tab1, tab2))
+
+        browserState = browserState.updateTabOrCustomTabState("tab1") {
+            it.createCopy(content = it.content.copy(url = updatedUrl))
+        }
+
+        assertEquals(2, browserState.customTabs.size)
+        assertEquals(updatedUrl, browserState.customTabs[0].content.url)
+        assertEquals(initialUrl, browserState.customTabs[1].content.url)
+    }
+
+    @Test
+    fun `GIVEN a private custom tab exists WHEN updateTabOrCustomTabState is called with the id of that tab THEN update the tab`() {
+        val tab1 = CustomTabSessionState(
+            id = "tab1", content = ContentState(url = initialUrl, private = true), config = mock()
+        )
+        val tab2 = CustomTabSessionState(
+            id = "tab2", content = ContentState(url = initialUrl, private = true), config = mock()
+        )
+        var browserState = BrowserState(customTabs = listOf(tab1, tab2))
+
+        browserState = browserState.updateTabOrCustomTabState("tab2") {
+            it.createCopy(content = it.content.copy(url = updatedUrl))
+        }
+
+        assertEquals(2, browserState.customTabs.size)
+        assertEquals(initialUrl, browserState.customTabs[0].content.url)
+        assertEquals(updatedUrl, browserState.customTabs[1].content.url)
+    }
+
+    @Test
+    fun `GIVEN tabs and custom tabs exist WHEN updateTabState is called with the id of the custom tab THEN the BrowserState is not updated`() {
```

Comment:
nit: THEN the **custom** tab is updated

Reply:
"THEN the custom tab is **not** updated" here because we're calling the cool new `updateTabState`.